karma-dashboard.config.cjs

```diff
@@ -57,7 +57,7 @@ module.exports = function (config) {
 
     // Frameworks to use.
     // Available frameworks: https://npmjs.org/browse/keyword/karma-adapter
-    frameworks: ['jasmine', 'karma-puppeteer-client'],
+    frameworks: ['jasmine', '@web-stories-wp/karma-puppeteer-client'],
```

Comment:
There's currently an error:

```
Server start failed on port 9876: Error: No provider for "framework:@web-stories-wp/karma-puppeteer-client"! (Resolving: framework:@web-stories-wp/karma-puppeteer-client)
```

We'll need to update `plugins` above too:

```diff
-require('./packages/karma-puppeteer-launcher/src/index.cjs'),
-require('./packages/karma-puppeteer-client/src/index.cjs'),
-require('./packages/karma-cuj-reporter/src/index.cjs'),
-require('./packages/karma-failed-tests-reporter/src/index.cjs'),
+require('@web-stories-wp/karma-puppeteer-launcher'),
+require('@web-stories-wp/karma-puppeteer-client'),
+require('@web-stories-wp/karma-cuj-reporter'),
+require('@web-stories-wp/karma-failed-tests-reporter'),
```

And of course add the packages to `package.json`. Then it should work.

Reply:
It still throws the same error

```bash
03 08 2021 14:27:21.283:ERROR [karma-server]: Server start failed on port 9877: Error: No provider for "framework:@web-stories-wp/karma-puppeteer-client"! (Resolving: framework:@web-stories-wp/karma-puppeteer-client)
03 08 2021 14:27:21.283:ERROR [karma-server]: Server start failed on port 9876: Error: No provider for "framework:@web-stories-wp/karma-puppeteer-client"! (Resolving: framework:@web-stories-wp/karma-puppeteer-client)
ERROR: "test:karma:edit-story" exited with 1.
```

Reply:
This was missing https://github.com/google/web-stories-wp/pull/8558/commits/cbe3846496174d47cdfbf8ff3bd584ae3a1b5720

Reply:
Hmm bummer.

According to the [docs](http://karma-runner.github.io/6.3/config/plugins.html) loading the plugins like that should work. It [even supports weildcards](https://github.com/karma-runner/karma/pull/3659) for that.

But might be missing something here 😕 Perhaps we have to dig into the plugin resolution a little bit to see where it's going wrong.

`node_modules/karma/lib/plugin.js` seems to be the file responsible for the loading of plugins.

Reply:
But exporting module key with the scope name solves the problem https://github.com/google/web-stories-wp/commit/cbe3846496174d47cdfbf8ff3bd584ae3a1b5720

Reply:
Woohoo!

Does this need to be done for the other packages' exports as well?

Reply:
I cannot find any documentation for that here https://karma-runner.github.io/5.2/dev/plugins.html
But what I can tell by looking at the examples of plugins is that the name of the export module should be the same as being used in reporters/framework/launchers array.

https://github.com/karma-runner/karma-growl-reporter/blob/master/index.js#L69
https://github.com/karma-runner/karma-growl-reporter/blob/master/README.md

```js
module.exports = function(config){
    config.set({
        reporters: ['progress','growl'],
   });
};
```

Currently we are using key names like 
```
'reporter:cuj'
'reporter:failed-tests'
'launcher:puppeteer'
```

I think we should keep it consistent with the package names.